components/cam/src/physics/crm/crm_physics.F90

```diff
@@ -729,25 +732,20 @@ subroutine crm_physics_tend(ztodt, state, tend, ptend, pbuf, cam_in, cam_out, &
       ! Set CRM inputs
       ! TODO: move this to a routine and call like:
       !    call set_crm_input(state, cam_in, pbuf, crm_input)
-      crm_input%zmid(1:ncol,1:pver) = state%zm(1:ncol,1:pver)
-      crm_input%zint(1:ncol,1:pver+1) = state%zi(1:ncol,1:pver+1)
-      crm_input%tl(1:ncol,1:pver) = state%t(1:ncol,1:pver)
-      crm_input%ql(1:ncol,1:pver) = state%q(1:ncol,1:pver,1)
-      crm_input%qccl(1:ncol,1:pver) = state%q(1:ncol,1:pver,ixcldliq)
-      crm_input%qiil(1:ncol,1:pver) = state%q(1:ncol,1:pver,ixcldice)
-      crm_input%ps(1:ncol) = state%ps(1:ncol)
-      crm_input%pmid(1:ncol,1:pver) = state%pmid(1:ncol,1:pver)
-      crm_input%pint(1:ncol,1:pver+1) = state%pint(1:ncol,1:pver+1)
-      crm_input%pdel(1:ncol,1:pver) = state%pdel(1:ncol,1:pver)
-      crm_input%phis(1:ncol) = state%phis(1:ncol)
-      crm_input%ul(1:ncol,1:pver) = state%u(1:ncol,1:pver)
-      crm_input%vl(1:ncol,1:pver) = state%v(1:ncol,1:pver)
-      crm_input%ocnfrac(1:ncol) = cam_in%ocnfrac(1:ncol)
-#if defined( MMF_ESMT )
-      ! Set the input wind for ESMT
-      crm_input%ul_esmt(1:ncol,1:pver) = state%u(1:ncol,1:pver)
-      crm_input%vl_esmt(1:ncol,1:pver) = state%v(1:ncol,1:pver)
-#endif /* MMF_ESMT */
+      crm_input_zmid(1:ncol,1:pver) = state%zm(1:ncol,1:pver)
+      crm_input_zint(1:ncol,1:pver+1) = state%zi(1:ncol,1:pver+1)
+      crm_input_tl(1:ncol,1:pver) = state%t(1:ncol,1:pver)
+      crm_input_ql(1:ncol,1:pver) = state%q(1:ncol,1:pver,1)
+      crm_input_qccl(1:ncol,1:pver) = state%q(1:ncol,1:pver,ixcldliq)
+      crm_input_qiil(1:ncol,1:pver) = state%q(1:ncol,1:pver,ixcldice)
+      crm_input_ps(1:ncol) = state%ps(1:ncol)
+      crm_input_pmid(1:ncol,1:pver) = state%pmid(1:ncol,1:pver)
+      crm_input_pint(1:ncol,1:pver+1) = state%pint(1:ncol,1:pver+1)
+      crm_input_pdel(1:ncol,1:pver) = state%pdel(1:ncol,1:pver)
+      crm_input_phis(1:ncol) = state%phis(1:ncol)
+      crm_input_ul(1:ncol,1:pver) = state%u(1:ncol,1:pver)
+      crm_input_vl(1:ncol,1:pver) = state%v(1:ncol,1:pver)
+      crm_input_ocnfrac(1:ncol) = cam_in%ocnfrac(1:ncol)
```

Comment:
I'm not sure we want to do this. We should have a discussion about how best to undo this input/output class suff.

Reply:
OpenMP4.5 doesn't support to map types/class at the moment, another way to do this is use similar implementation as crm_state by creating new variables inside crm_module code. This way will keep the crm_physics and crm_history code nice. @whannah1 

I will try to remove some of the atomic update, but I have found that OpenMP is different from OpenACC in the way of atomic update, it turns out requiring more atomic update than OpenACC. @mrnorman